### PR TITLE
Fix foreign derive proc macro attributes clashing with Veil's

### PR DESCRIPTION
**Pull request by WilliamVenner**
Spotted this issue in the wild

### CODE REVIEW

File: veil-tests/src/compile_tests.rs [hunk 1]
```diff
@@ -183,4 +183,13 @@ fn test_redact_multiple_attributes() {
...
     }
+
+    #[derive(serde::Serialize, Redact)]
+    #[serde(rename_all = "camelCase")]
+    #[redact(all)]
+    struct MultipleAttributesAll {
+        #[serde(default)]
+        hidden: bool,
+        exposed: bool,
+    }
```
Comment (MaeIsBad) [inline veil-tests/src/compile_tests.rs:194]: Should these tests also verify that serde attributes still get applied? Or maybe doing this in a separate integration test
```suggestion
    let json = serde_json::json!({
        "exposed": true
    });
    let attributes: MultipleAttributesAll = serde_json::from_value(json).unwrap();
    assert_eq!(attributes.exposed, true);
    assert_eq!(attributes.hidden, false);
}
```
(this suggestion won't compile as it needs serde_json as well)

File: veil-macros/src/sanitize.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+trait AttributeFilter {
```
Comment (MaeIsBad) [inline veil-macros/src/sanitize.rs:1]: I like this implementation, but I feel like this might have some edge cases, so maybe we should write some tests here? If you don't feel like doing it I'd be ok with writing them myself

Comment (WilliamVenner) [reply veil-macros/src/sanitize.rs:1]: Actually found some bugs thx to this :)

**Review by MaeIsBad — APPROVED** [review]
LGTM
